goto_commands.py

```diff
@@ -0,0 +1,136 @@
+    "command": "sublime_linter_goto_error",
+    "args": {"direction": "previous"}
+  },
+```
+
+Supported args:
+
+* `direction`: 'previous' | 'next'
+* `count`: How wide you will jump. Defaults to 1.
+* `wrap`: Set to True, to jump to the top if you're on the last error and
+          vice versa. Defaults to False
+
+"""
+
+
+class SublimeLinterGotoError(sublime_plugin.WindowCommand):
+    def run(self, direction='next', count=1, wrap=False):
+        goto(self.window.active_view(), direction, count, wrap)
+
+
+def goto(view, direction, count, wrap):
+    bid = view.buffer_id()
+
+    try:
+        errors = persist.raw_errors[bid]
+    except KeyError:
+        flash(view, 'No errors/problems')
+        return
+
+    if len(errors) == 0:
+        flash(view, 'No errors/problems')
+        return
+
+    current_line, current_col = get_current_pos(view)
+    if (current_line, current_col) == (-1, -1):
+        return
+
+    # Filter out duplicates on the same position. Also filter out errors
+    # under the cursor, bc we don't want to jump to them.
+    errors = {(error['line'], error['start'])
+              for error in errors
+              if not(error['line'] == current_line and
+                     error['start'] <= current_col <= error['end'])}
+    errors = sorted(errors)
+
+    def before_current_pos(error):
+        line, start = error
+        return (
+            line < current_line or
+            # `start < current_col` is only safe bc we already
+            # filtered errors directly under cursor (!)
+            line == current_line and start < current_col)
+
+    next = dropwhile(before_current_pos, errors)
+    previous = takewhile(before_current_pos, errors)
+
+    reverse = direction == 'previous'
+    jump_positions = list(previous if reverse else next)
+
+    if not jump_positions:
+        if wrap:
+            error = errors[-1] if reverse else errors[0]
+        else:
+            flash(
+                view,
+                'No more problems {}.'.format('above' if reverse else 'below')
+            )
+            return
+    elif len(jump_positions) <= count:
+        # If we cannot jump wide enough, pin to top/bottom to reduce
+        # disorientation.
+        error = errors[0] if reverse else errors[-1]
+    else:
+        # This is a bit hard to read, but note that for any list we get the
+        # *first* item via `list[0]`, but going reverse, we get the *last*
+        # item via `list[-1]` (not via `list[-0]`).
+        error = jump_positions[-count] if reverse else jump_positions[count - 1]
+
+    line, start = error
+    move_to(view, line, start)
+
+
+def move_to(view, line, col):
+    loc = view.text_point(line, col)
+    region = sublime.Region(loc, loc)
+    view.sel().clear()
+    view.sel().add(region)
+    center_region_in_view(view, region)
+
+
+def center_region_in_view(view, region):
+    """Center the given region in view.
+
+    There is a bug in ST3 that prevents a selection change
```

Comment:
It seems the currently preferred way for this is to perform selection within in a text command so they can be soft-undone.

Reply:
I don't know how to do that. I thought about this, but failed bc there is no documentation of what commands Sublime provides. 

You could fix this in a separate PR.

Reply:
Rechecked, I can jump back via `alt-left` just fine.

Reply:
I think you'd need your own command for this.

Reply:
Otoh, you could try View.open_file with `sublime.ENCODED_POSITION`.

Reply:
Okay, got it